sdk/translation/azure-ai-translation-text/azure/ai/translation/text/aio/_patch.py

```diff
@@ -0,0 +1,45 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+from typing import Union
+from azure.core.credentials import ( TokenCredential, AzureKeyCredential )
+
+from .._patch import (
+    get_translation_endpoint,
+    set_authentication_policy,
+    TranslatorCredential
+)
+
+from ._client import TextTranslationClient as ServiceClientGenerated
+
+def patch_sdk():
+    """Do not remove from this file.
+
+    `patch_sdk` is a last resort escape hatch that allows you to do customizations
+    you can't accomplish using the techniques described in
+    https://aka.ms/azsdk/python/dpcodegen/python/customize
+    """
+
+class TextTranslationClient(ServiceClientGenerated):
+    def __init__(
+            self,
+            endpoint: Union[str , None],
+            credential: Union[AzureKeyCredential , TokenCredential , TranslatorCredential],
+            **kwargs):
+
+        set_authentication_policy(credential, kwargs)
```

Comment:
We'll need to customize this for async since the token auth has a different async BearerTokenPolicy. See here for an example: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/_patch.py#L24

Reply:
Done